system-x/services/splunk/src/main/java/software/tnb/splunk/service/openshift/Splunk.java

```diff
@@ -0,0 +1,220 @@
+package software.tnb.splunk.service.openshift;
+
+import software.tnb.common.account.AccountFactory;
+import software.tnb.common.config.OpenshiftConfiguration;
+import software.tnb.common.deployment.OpenshiftDeployable;
+import software.tnb.common.deployment.WithExternalHostname;
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.service.Service;
+import software.tnb.common.utils.WaitUtils;
+import software.tnb.splunk.account.SplunkAccount;
+import software.tnb.splunk.validation.SplunkValidation;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.service.AutoService;
+import com.google.common.io.Resources;
+import com.splunk.ServiceArgs;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import cz.xtf.core.openshift.helpers.ResourceParsers;
+import io.fabric8.kubernetes.api.model.DeletionPropagation;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRole;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.api.model.RouteBuilder;
+
+@AutoService(Splunk.class)
+public class Splunk implements Service, OpenshiftDeployable, WithExternalHostname {
+    private static final Logger LOG = LoggerFactory.getLogger(Splunk.class);
+    public static final String ROUTE_NAME = "api";
+    public static final int API_PORT = 443;
+    private SplunkAccount account;
+    private com.splunk.Service client;
+    private SplunkValidation validation;
+    private List<HasMetadata> createdResources;
+    private CustomResourceDefinitionContext crdContext;
+    private Route apiRoute;
+
+    private static final String SCC_NAME = "tnb-splunk";
+
+    public SplunkAccount account() {
+        if (account == null) {
+            LOG.debug("Creating new Splunk account");
+            account = AccountFactory.create(SplunkAccount.class);
+            account.setPassword(
+                new String(Base64.getDecoder().decode(OpenshiftClient.get().getSecret("splunk-s1-standalone-secret-v1").getData().get("password"))));
+        }
+        return account;
+    }
+
+    /**
+     * Due to self sign certificate, the client is not able to communicate via localhost and port-forward.
+     * OCP external route with `reencrypt` is used. (Cluster needs to have valid certificate!)
+     */
+    protected com.splunk.Service client() {
+        LOG.debug("Creating new Splunk client");
```

Comment:
wrap in `if (client == null)` if it is not necessary to create a new client everytime